load_balancer/load_balancer.js

```diff
@@ -320,62 +288,29 @@ try {
 } finally {
 
 }
-var node_queue=[];
-for(var i=0;i<nodes_data.Nodes.length;i++)
-{
-  checkNodeConn(nodes_data.Nodes[i]);
-  function checkNodeConn(node) {
 
-    var https_checkConn ={
-      hostname : node.hostname,
-      port : node.port,
-      path : '/connectionCheck',
-      key : fs.readFileSync('./ssl/key.pem'),
-      cert: fs.readFileSync('./ssl/cert.pem'),
-      rejectUnauthorized:false,
-    };
-
-  var checkConnRequest = https.request(https_checkConn,function(res)
-  {
-    var bodyChunks =[];
-    res.on('data',function(chunk)
-    {
-      bodyChunks.push(chunk);
-    }).on('end',function()
-    {
-      var body = Buffer.concat(bodyChunks);
-      if(body.toString()=='true')
-      {
-        console.log("Added "+node.hostname+":"+node.port+" to queue");
-        node_queue.push(node);
-      }
-    });
-  });
-
-  checkConnRequest.on('error',function(err)
-{
-  console.log("Error connecting to "+node.hostname+":"+node.port);
+/* This will update the node queue and working nodes will be added.
+The logger level would be info, when logger.js is integrated. */
 
+status.selectActiveNodes().then((workingNodes) => {
+  node_queue = Object.assign([], workingNodes);
+}).catch((err) => {
+  console.log(err);
 });
-checkConnRequest.end();
-
-  }
-}
 
 var job_queue = [];
+
 if(process.env.mode !== "TESTING")
 {
-  server.listen(nodes_data.host_port.port);
-  console.log("Listening at "+nodes_data.host_port.port);
+  server.listen(nodes_data.load_balancer.port);
+  console.log("Listening at "+nodes_data.load_balancer.port);
 }
 
-
-
 setInterval(function () {
   connection.query('SELECT 1' ,function(err, rows, fields) {
     console.log("keep alive query");
   });
-}, 10000);
+}, 100000);
```

Comment:
The timeout value here would be half of the socket timeout set in MySQL config file.

Reply:
The default value for timeout is 8 hrs. We can have out own custom timeout value with the wait_timeout variable. It was verified and here is the same.
`````
mysql> SHOW global VARIABLES like "wait_timeout";
+---------------+-------+
| Variable_name | Value |
+---------------+-------+
| wait_timeout  | 28800 |
+---------------+-------+
1 row in set (0.07 sec)
`````      

We can set the timeout value to 14400 here.